src/phongo_client.c

```diff
@@ -51,6 +51,47 @@ typedef struct {
 	bool             is_persistent;
 } php_phongo_pclient_t;
 
+static const mongoc_client_t* get_first_pclient_client(HashTable* ht)
+{
+	if (ht) {
+		php_phongo_pclient_t* pclient = NULL;
+
+		zend_hash_internal_pointer_reset(ht);
+		pclient = zend_hash_get_current_data_ptr(ht);
+
+		if (pclient) {
+			return pclient->client;
+		}
+	}
+
+	return NULL;
+}
+
+/* Returns the version of the crypt_shared library, or NULL if it's unavailable.
+ * Querying the version requires a mongoc_client_t pointer. Since the shared
+ * library can only be loaded once, any client will return the same result so we
+ * consult the first persistent or request-scoped client we can find.
+ *
+ * Note: this may incorrectly return NULL if crypt_shared was loaded through a
+ * mongoc_client_t since destroyed (e.g. single requested-scoped client);
+ * however, that's the best can do with libmongoc's API. */
```

Comment:
If I read this correctly, then we wouldn't report a version if calling something like `php --ri mongodb`, since that never instantiates a client - is that correct? Is there a way to work around this limitation?

Reply:
It goes beyond that. `crypt_shared` is only loaded dynamically as needed, so if the Manager objects are not constructed with `autoEncryption` driver options that would necessitate loading the library, `_mongoc_crypt_get_crypt_shared_version` will return NULL.

We could theoretically try to get around that by constructing an extra client (without `cryptSharedLibRequired` to avoid errors), but there's no guarantee that it would pick up the same library or even a library at all since we don't know what path a user might specify in `cryptSharedLibPath`. We would be limited to searching the system path.

Reply:
I see. No worries then, and "unknown" is vague enough to cover those cases.